nanofi/include/core/log.h

```diff
@@ -0,0 +1,76 @@
+/**
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef NIFI_MINIFI_CPP_LOG_H
+#define NIFI_MINIFI_CPP_LOG_H
+
+#include <stdio.h>
+#include <stdlib.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef enum
+{
+  off = 0,
+  critical = 1,
+  err = 2,
+  warn = 3,
+  info = 4,
+  debug = 5,
+  trace = 6,
+} log_level;
+
+extern int global_log_level;
```

Comment:
may want to hint to the optimizer with volatile, generally this is a single CPU instruction but I wouldn't guarantee that to always be the case.

Reply:
Good shout, added, thanks!